src/components/profile.ts

```diff
@@ -142,15 +151,20 @@ export const getUserProfileById = async (userId: string): Promise<UserProfile |
 
 // NOTE: data will always be just one customization, but we do not know which one, meaning many of
 // the loops are just to extract an unknown key-val pair
-export const editUserProfileById = async (userId: string, data: UserProfile): Promise<void> => {
+export const editUserProfileById = async (userId: string, data: UserProfile, message: Message): Promise<void> => {
+  // grab oldYear as that might be used later on when updating year roles
+  const oldProfileDetails: UserProfile | undefined = await getUserProfileById(userId);
+  const oldYear = oldProfileDetails?.year || null;
```

Comment:
Can this be moved into `updateProfileRoles`, so `oldYear` also doesn't have to be a parameter?